test/Test/Toml/Parsing/Unit.hs

```diff
@@ -0,0 +1,308 @@
+        parseInt "+0" 0
+        parseInt "-0" 0
+      it "can parse both the minimum and maximum numbers in the 64 bit range" $ do
+        parseInt "-9223372036854775808" (-9223372036854775808)
+        parseInt "9223372036854775807" 9223372036854775807
+      xit "can parse numbers with underscores between digits" $ do
+        parseInt "1_000" 1000
+        parseInt "5_349_221" 5349221
+        parseInt "1_2_3_4_5" 12345
+        parseInt "1_2_3_" 1
+        parseInt "13_" 13
+        intFailOn "_123_"
+        intFailOn "_13"
+        intFailOn "_"
+      xit "does not parse numbers with leading zeros" $ do
+        parseInt "0123" 0
+        parseInt "-023" 0
+    context "when the integer is in binary representation" $ do
+      it "can parse numbers prefixed with `0b`" $ do
+        parseInt "0b1101" 13
+        parseInt "0b0" 0
+      it "does not parse numbers prefixed with `0B`" $ do
+        parseInt "0B1101" 0
+      it "can parse numbers with leading zeros after the prefix" $ do
+        parseInt "0b000" 0
+        parseInt "0b00011" 3
+      it "does not parse negative numbers" $ do
+        parseInt "-0b101" 0
+      it "does not parse numbers with non-valid binary digits" $ do
+        parseInt "0b123" 1
+    context "when the integer is in octal representation" $ do
+      it "can parse numbers prefixed with `0o`" $ do
+        parseInt "0o567" 0o567
+        parseInt "0o0" 0
+      it "does not parse numbers prefixed with `0O`" $ do
+        parseInt "0O567" 0
+      it "can parse numbers with leading zeros after the prefix" $ do
+        parseInt "0o000000" 0
+        parseInt "0o000567" 0o567
+      it "does not parse negative numbers" $ do
+        parseInt "-0o123" 0
+      it "does not parse numbers with non-valid octal digits" $ do
+        parseInt "0o789" 0o7
+    context "when the integer is in hexadecimal representation" $ do
+      it "can parse numbers prefixed with `0x`" $ do
+        parseInt "0x12af" 0x12af
+        parseInt "0x0" 0
+      it "does not parse numbers prefixed with `0X`" $ do
+        parseInt "0Xfff" 0
+      it "can parse numbers with leading zeros after the prefix" $ do
+        parseInt "0x00000" 0
+        parseInt "0x012af" 0x12af
+      it "does not parse negative numbers" $ do
+        parseInt "-0xfff" 0
+      it "does not parse numbers with non-valid hexadecimal digits" $ do
+        parseInt "0xfgh" 0xf
+      it "can parse numbers when hex digits are lowercase" $ do
+        parseInt "0xabcdef" 0xabcdef
+      it "can parse numbers when hex digits are uppercase" $ do
+        parseInt "0xABCDEF" 0xABCDEF
+      it "can parse numbers when hex digits are in both lowercase and uppercase" $ do
+        parseInt "0xAbCdEf" 0xAbCdEf
+        parseInt "0xaBcDeF" 0xaBcDeF
+
+  describe "keyP" $ do
+    context "when the key is a bare key" $ do
+      it "can parse keys which contain ASCII letters, digits, underscores, and dashes" $ do
+        parseKey "key"       (makeKey ["key"])
```

Comment:
Actually, `makeKey` function is not needed since there exist `IsString` instance for key. So these tests could be written just like

```haskell
parseKey "key" "key"
```

But types of first and second arguments are different, so that's actually okay! If you prefer explicit style more, you can instead write:

```haskell
parseKey "key" ("key" :: Key)
```

Reply:
For this case:
```haskell
parseKey "site.\"google.com\"" "site.\"google.com\""
```
I get the following:
```
expected: Key {unKey = Piece {unPiece = "site"} :| [Piece {unPiece = "\"google"},Piece {unPiece = "com\""}]}
but got: Key {unKey = Piece {unPiece = "site"} :| [Piece {unPiece = "\"google.com\""}]}
```
`\"google.com\"` should be a single `Piece` and should not be split.

I've seen that this issue in `IsString` implementation was addressed here:
https://github.com/kowainik/tomland/blob/533e2d5efccb8e38922d38f116cded3c35a81be0/src/Toml/PrefixTree.hs#L67-L71

What do you suggest to do in this case?

Reply:
@ghallak Ah, right, `IsString` instance is not supposed to work with quoted keys.

Reply:
So should I keep using `makeKey` ?

Reply:
@ghallak It's on your choice now. In that case I'm okay if you will use `makeKey` to make tests look more consistent. But you also can use `makeKey` only for quoted keys.

Reply:
@ChShersh Will go with consistency this time. Using `makeKey` only for dotted keys which have quoted keys as part of them, will make things look strange.